server.go

```diff
@@ -1512,6 +1508,31 @@ func randomUint16Number(max uint16) uint16 {
 	}
 }
 
+func (s *server) attemptDcrdDial(ctx context.Context, network, addr string) (net.Conn, error) {
+	host, portStr, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, err
+	}
+	port, err := strconv.ParseUint(portStr, 10, 16)
+	if err != nil {
+		return nil, err
+	}
+	remoteAddr, err := s.addrManager.HostToNetAddress(host, uint16(port), 0)
+	if err != nil {
+		return nil, err
+	}
+	// Be sure the address exists in the address manager.
+	s.addrManager.AddAddresses([]*addrmgr.NetAddress{remoteAddr},
```

Comment:
This will now end up doing this (and the attempt below) when it isn't supposed to (e.g. simnet and regnet).